src/gc/gc.cpp

```diff
@@ -3979,7 +3979,7 @@ class CObjectHeader : public Object
 #endif
         if (fSmallObjectHeapPtr)
         {
-#ifdef FEATURE_BASICFREEZE
+#if defined(FEATURE_BASICFREEZE) && defined(VERIFY_HEAP)
```

Comment:
The change in this ifdef does not look right. Why is it necessary?

Reply:
IsInFrozenSegment is only defined when Verify Heap is enabled.

Reply:
But I can improve the assert condition if that's what you were getting at? I can make it this --

```cpp
#if defined(FEATURE_BASICFREEZE) && defined(VERIFY_HEAP)
            _ASSERTE(g_theGCHeap->IsInFrozenSegment(this));
#endif
            _ASSERTE(!g_theGCHeap->IsLargeObject(pMT));
```

Reply:
Probably should, updated.

Reply:
@Maoni0 does this look correct?